packages/dapp-kit-next/src/store/initializers/registered-wallets.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getWallets, StandardEvents } from '@mysten/wallet-standard';
+import { onMount } from 'nanostores';
+import type { DAppKitState } from '../state.js';
+
+import type { Wallet, WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+import { getSuiWallets, isSuiWallet } from '../../utils/wallets.js';
+import { getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED as getOrCreateUiWalletForStandardWallet } from '@wallet-standard/ui-registry';
+
+/**
+ * Handles updating the store in response to wallets being added, removed, and their properties changing.
+ */
+export function syncRegisteredWallets({ $state }: DAppKitState) {
+	onMount($state, () => {
+		const walletsApi = getWallets();
+		const unsubscribeCallbacksByWallet = new Map<Wallet, () => void>();
+
+		const onWalletsChanged = () => {
+			const suiWallets = getSuiWallets().map(getOrCreateUiWalletForStandardWallet);
+			$state.setKey('wallets', suiWallets);
+		};
+
+		const subscribeToWalletEvents = (wallet: WalletWithRequiredFeatures) => {
```

Comment:
You could also subscribe to changes on $state.wallets and do this there if you wanted to separate these things out (wallets changing vs wallets being added / removed), but if not this is also cool